lib/Doctrine/Common/Cache/RiakCache.php

```diff
@@ -19,69 +19,87 @@
 
 namespace Doctrine\Common\Cache;
 
-use Riak\Bucket;
-use Riak\Input;
-use Riak\Exception;
-use Riak\Object;
+use Basho\Riak;
+use Basho\Riak\Command;
+use Basho\Riak\Exception as RiakException;
+use Basho\Riak\Object;
 
 /**
  * Riak cache provider.
  *
  * @link   www.doctrine-project.org
  * @since  1.1
  * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
+ * @author Anthon Pang <apang@softwaredevelopment.ca>
  */
 class RiakCache extends CacheProvider
 {
     const EXPIRES_HEADER = 'X-Riak-Meta-Expires';
 
     /**
-     * @var \Riak\Bucket
+     * @var \Basho\Riak
      */
-    private $bucket;
+    private $riak;
+
+    /**
+     * @var string
+     */
+    private $bucketName;
 
     /**
      * Sets the riak bucket instance to use.
      *
-     * @param \Riak\Bucket $bucket
+     * @param \Basho\Riak $riak
+     * @param string      $bucketName
      */
-    public function __construct(Bucket $bucket)
+    public function __construct(Riak $riak, $bucketName)
```

Comment:
This is a major BC break: please copy and create a new class instead, and deprecate the existing client instead.